v3/onesignal-admin/onesignal-admin.php

```diff
@@ -22,21 +22,25 @@ function admin_files()
 
 if ($_SERVER['REQUEST_METHOD'] === 'POST') {
   if (isset($_POST["submit"])) {
+      $onesignal_settings = get_option('OneSignalWPSetting', array());
 
-    $onesignal_settings = get_option('OneSignalWPSetting', array());
+      if (isset($_POST['onesignal_app_id']) && !empty($_POST['onesignal_app_id'])) {
+          $onesignal_settings['app_id'] = sanitize_text_field($_POST['onesignal_app_id']);
+      }
 
-    if (isset($_POST['onesignal_app_id']) && !empty($_POST['onesignal_app_id'])) {
-      $onesignal_settings['app_id'] = sanitize_text_field($_POST['onesignal_app_id']);
-    }
+      if (isset($_POST['onesignal_rest_api_key']) && !empty($_POST['onesignal_rest_api_key'])) {
+          $onesignal_settings['app_rest_api_key'] = sanitize_text_field($_POST['onesignal_rest_api_key']);
+      }
 
-    if (isset($_POST['onesignal_rest_api_key']) && !empty($_POST['onesignal_rest_api_key'])) {
-      $onesignal_settings['app_rest_api_key'] = sanitize_text_field($_POST['onesignal_rest_api_key']);
-    }
+      // Save the auto send notifications setting
+      $auto_send = isset($_POST['onesignal_auto_send']) ? 1 : 0;
```

Comment:
Question: Is it enough to check that `onesignal_auto_send` exists here, or should we ensure that its value is set to 1?

Reply:
Good question. It won't come through at all from the POST action if unchecked.